### ISSUE TITLE
Breakage: nested pages have wrong links

### ISSUE THREAD

**Issue by steveklabnik** [azerupi/mdBook#417]
Originally filed as https://github.com/azerupi/mdBook/issues/412

the unstable book uses nested pages, and headers aren't generating the correct links, that is, unstable-book\library-features\windows-stdio.html:89: broken link - unstable-book\windows-stdio.html (note the missing library-features path.

When using a book that has contents in subdirectories, headers used to properly generate links, but instead, don't have the subdirectory name.

In other words, if the file is `library-features\windows-stdio.md`, the link is generated as `windows-stdio.html` rather than `library-features\windows-stdio.html`, missing the subdirectory.

Sample book here: https://github.com/steveklabnik/mdbook-bug-report/tree/master/417

**Comment by behnam** [comment]
You beat me to it, @steveklabnik! I noticed this on the Cargo mdBook docs, like this page: http://doc.crates.io/book/reference/publishing.html

Generally, I would expect all in-page links to be fragment-only URLs (starting with `#`). I'm surprised that the `href` value actually has non-fragment parts to begin with. Curious if there's any practical reason for it, or it's just because of some internal details.

Anyways, now need to fix this before we can publish Cargo book.
